libs/core/routing/src/uri/normalizer.service.ts

```diff
@@ -0,0 +1,37 @@
+import { Injectable } from '@angular/core';
+import {
+  UrlSerializer,
+  UrlTree,
+  PRIMARY_OUTLET,
+} from '@angular/router';
+
+/**
+ * Processes URIs to remove extraneous Angular-specific information.
+ */
+@Injectable({
+  providedIn: 'root',
+})
+export class DaffRoutingUriNormalizer {
+  constructor(
+    private urlParser: UrlSerializer,
+  ) {}
+
+  /**
+   * Normalizes an Angular router path into a useable URI.
+   * e.g. `normalize('some/url.html(secondary:outlet)?query=param#fragment')` -> `'some/url.html?query=param#fragment'`
+   *
+   * Optionally accepts an outlet to target.
+   * e.g. `normalize('some/url.html(secondary:outlet)?query=param#fragment', 'secondary')` -> `'some/outlet?query=param#fragment'`
```

Comment:
This doc isn't correct given the test.

Reply:
yeah good call. I actually didn't realize that it would remove leading path segments when targeting a different outlet, is this going to be a problem?

Reply:
I don't think so.